test_pull_requests

```diff
@@ -2089,7 +2089,8 @@ popd
       # First, get the next jenkins build number
       next_build_num = get_next_build(branch, settings)
 
-      pull_id_params = "&BRANCH=#{branch}"
+      branch_param = settings["branch_param"] || "BRANCH"
```

Comment:
But we have no concept of config sharing between triggers today... our JSON is like 99% duplicated because of that. I agree that this is not ideal but that's the architecture here, isn't it?